File: src/main/scala/com/wordnik/swagger/codegen/util/ValidationException.scala
package com.wordnik.swagger.util

import com.wordnik.swagger.codegen.model._

import scala.collection.JavaConverters._
import scala.beans.BeanProperty

class ValidationException(code:Int, msg:String, errors: List[ValidationError]) extends Exception(msg:String) {
  val messages: java.util.List[ValidationMessage] = (
    for(e <- errors) yield ({
      val m = new ValidationMessage()
      m.path = e.path
      m.message = e.message
      m.severity = e.severity
      m
    })
  ).toList.asJava
}

class ValidationMessage() {
  @BeanProperty var path: String = _
  @BeanProperty var message: String = _
  @BeanProperty var severity: String = _
}

File: modules/swagger-generator/src/main/java/io/swagger/generator/util/ValidationException.java
package io.swagger.generator.util;

public class ValidationException extends Exception {
    private static final long serialVersionUID = 6861195361018260380L;
    private int code;
    private String msg;

    public ValidationException(String msg) {
        super(msg);
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    @Override
    public String getMessage() {
        return msg;
    }

    public void setMessage(String msg) {
        this.msg = msg;
    }
}